### PR TITLE
Support for downloading archive of R documentation from midas

### PR DESCRIPTION
**Pull request by richardbeare**
Change-Id: Ib713a817839e5122d0f8ccdf3cbcd808c2ec4d8d

This change fetches a tar.gz of automatically generated R documentation files and extracts them into the package build hierarchy.

### CODE REVIEW

File: Wrapping/R/CMakeLists.txt [hunk 1]
```diff
@@ -67,7 +67,7 @@ configure_file(
 if(NOT SimpleITK_FORBID_DOWNLOADS)
   include(sitkExternalData)
 
-  file( GLOB_RECURSE content_links RELATIVE "${CMAKE_CURRENT_SOURCE_DIR}" "*.md5" )
+  file( GLOB_RECURSE content_links RELATIVE "${CMAKE_CURRENT_SOURCE_DIR}" "Packaging/SimpleITK/data/*.md5" )
```
Comment (blowekamp) [inline Wrapping/R/CMakeLists.txt]: Please Specify the last argument as "${CMAKE_CURRENT_SOURCE_DIR}/Packaging/SimpleITK/data/*.md5"

File: Wrapping/R/CMakeLists.txt [hunk 2]
```diff
@@ -91,6 +91,33 @@ if(NOT SimpleITK_FORBID_DOWNLOADS)
     )
 
 endif()
+# Download tar of Rd files
+if(NOT SimpleITK_FORBID_DOWNLOADS)
+  include(sitkExternalData)
+
+  file( GLOB_RECURSE content_links RELATIVE "${CMAKE_CURRENT_SOURCE_DIR}" "Packaging/SimpleITK/man/*.md5" )
```
Comment (blowekamp) [inline Wrapping/R/CMakeLists.txt]: Same explicitly for the last argument here too.

File: Wrapping/R/CMakeLists.txt [hunk 3]
```diff
@@ -91,6 +91,33 @@ if(NOT SimpleITK_FORBID_DOWNLOADS)
     )
 
 endif()
+# Download tar of Rd files
+if(NOT SimpleITK_FORBID_DOWNLOADS)
+  include(sitkExternalData)
```
Comment (blowekamp) [inline Wrapping/R/CMakeLists.txt]: This is duplication from above, I think the code block below and above should be consolidated, with just one SimpleITKRData target.

File: Wrapping/R/Packaging/SimpleITK/man/sitkRdfiles.tar.gz.md5 [hunk 4]
```diff
@@ -0,0 +1 @@
+a3a0acaeba6877ebf40f866d9e1c54ac
```
Comment (blowekamp) [inline Wrapping/R/Packaging/SimpleITK/man/sitkRdfiles.tar.gz.md5]: I have also uploaded the file to the ITK/SimpleITK folder.

File: Wrapping/R/CMakeLists.txt (hunk 3 above)
Comment (richardbeare) [inline Wrapping/R/CMakeLists.txt]: I separated as the actions after download are different - copy vs tar extraction. Is there a good way to separate the two if the download code is common? Using filenames seems just as complicated, but feasible

Comment (richardbeare) [inline Wrapping/R/CMakeLists.txt]: Just realised you probably mean to use a single if statement

Comment (blowekamp) [inline Wrapping/R/CMakeLists.txt]: yes, I meant one if statement. You can add multiple commands to the PRE_BUILD step of the target to handle the two different types.

File: Wrapping/R/CMakeLists.txt [hunk 5]
```diff
@@ -89,10 +102,13 @@ if(NOT SimpleITK_FORBID_DOWNLOADS)
     PRE_BUILD
     ${COPY_DATA_COMMAND}
```
Comment (blowekamp) [inline Wrapping/R/CMakeLists.txt]: Can you add ${EXTRACT_RMAN_COMMAND} here. Then remove the second add_custom_command.

I think that the PRE_BUILD command will be overwritten with the second statement without the APPEND option[1]. Simplifying, to a single add_custom_command should clear that caveat up.
[1] https://cmake.org/cmake/help/v3.0/command/add_custom_command.html

**Comment by blowekamp** [comment]
Thank you I have rebased the commit onto the current master and made another pull request here #130.

This patch appears to have been based on next. Please base all pull request on master.

**Pull request closed without merging**